src/query/parser/promql/parse.go

```diff
@@ -266,6 +274,19 @@ func (p *parseState) walk(node pql.Node) error {
 		// Evaluate inside of paren expressions
 		return p.walk(n.Expr)
 
+	case *pql.UnaryExpr:
+		err := p.walk(n.Expr)
+		if err != nil {
+			return err
+		}
+
+		unaryType, ok := GetUnaryOpType(n.Op)
+		if !ok {
+			return fmt.Errorf("only + and - operators allowed for unary expressions, received: %s", n.Op.String())
+		}
+
+		return p.addLazyTransform(0, lazy.UnaryType, unaryType)
```

Comment:
This is a bit janky; I think it's better to split the functions up for now (i.e. noops if it's plus, and creates a value transform otherwise)